src/otx/engine/engine.py

```diff
@@ -778,6 +779,80 @@ def explain(
         model.explain_mode = False
         return predict_result
 
+    def benchmark(
+        self,
+        checkpoint: PathLike | None = None,
+        batch_size: int = 1,
+        n_iters: int = 10,
+        extended_stats: bool = False,
+    ) -> dict[str, str]:
+        checkpoint = checkpoint if checkpoint is not None else self.checkpoint
+
+        if checkpoint is not None:
+            is_ir_ckpt = Path(checkpoint).suffix in [".xml"]
+            if is_ir_ckpt and not isinstance(self.model, OVModel):
+                # create OVModel
+                self.model = self._auto_configurator.get_ov_model(
+                    model_name=str(checkpoint),
+                    label_info=self.datamodule.label_info,
+                )
+
+            if not is_ir_ckpt:
+                model_cls = self.model.__class__
+                self.model = model_cls.load_from_checkpoint(
+                    checkpoint_path=checkpoint,
+                    map_location="cpu",
+                    **self.model.hparams,
+                )
+        elif isinstance(self.model, OVModel):
+            msg = "To run benchmark on OV model, checkpoint must be specified."
+            raise RuntimeError(msg)
+
+        self.model.eval()
+
+        def dummy_infer(model: OTXModel, batch_size: int = 1, extra_stats: bool = False) -> dict[str, Any]:
+            input_batch = model.get_dummy_input(batch_size)
+            stats = {}
+            start = time.time()
+            if extra_stats:
+                from torch.utils.flop_counter import get_suffix_str, convert_num_with_suffix
+
+                model_fwd = lambda: model.forward(input_batch)
+                depth = 3 if extended_stats else 0
+                fwd_flops = measure_flops(model.model, model_fwd, print_stats_depth=depth)
```

Comment:
Is it ok to measure flops while mesureing speed? Doesn't it spend more time than just executing forward?

Reply:
I the benchmark scenario I estimate flops with a standalone call, but overall flops counter is a bunch of lightweight hooks on top of torch operations. I'll think about refactoring.